packages/ethereum-storage/src/ethereum-storage.ts

```diff
@@ -182,47 +197,60 @@ export default class EthereumStorage implements StorageTypes.IStorage {
     }
 
     // Add content to IPFS and get the hash back
-    let ipfsHash;
+    let ipfsHash: string;
     try {
       ipfsHash = await this.ipfsManager.add(content);
     } catch (error) {
       throw Error(`Ipfs add request error: ${error}`);
     }
 
     // Get content length from ipfs
-    let contentSize;
+    let contentSize: number;
     try {
       contentSize = await this.ipfsManager.getContentLength(ipfsHash);
     } catch (error) {
       throw Error(`Ipfs get length request error: ${error}`);
     }
 
-    const feesParameters: StorageTypes.IFeesParameters = { contentSize };
-
-    // Add content hash to ethereum
-    let ethereumMetadata;
-    try {
-      ethereumMetadata = await this.smartContractManager.addHashAndSizeToEthereum(
-        ipfsHash,
-        feesParameters,
-      );
-    } catch (error) {
-      throw Error(`Smart contract error: ${error}`);
-    }
-
-    // Save the metadata of the new ipfsHash into the Ethereum metadata cache
-    await this.ethereumMetadataCache.saveDataIdMeta(ipfsHash, ethereumMetadata);
-
-    return {
+    const timestamp = Utils.getCurrentTimestampInSecond();
+    const result: StorageTypes.IAppendResult = Object.assign(new EventEmitter(), {
       content,
       id: ipfsHash,
       meta: {
-        ethereum: ethereumMetadata,
         ipfs: { size: contentSize },
-        storageType: StorageTypes.StorageSystemType.ETHEREUM_IPFS,
-        timestamp: ethereumMetadata.blockTimestamp,
+        local: { location: this.locationBufferUrl },
+        storageType: StorageTypes.StorageSystemType.LOCAL,
+        timestamp,
       },
-    };
+    });
+    // store in the buffer the timestamp
+    this.buffer[ipfsHash] = timestamp;
+
+    const feesParameters: StorageTypes.IFeesParameters = { contentSize };
+
+    this.smartContractManager
+      .addHashAndSizeToEthereum(ipfsHash, feesParameters)
+      .then(async (ethereumMetadata: StorageTypes.IEthereumMetadata) => {
+        const resultAfterBroadCast: StorageTypes.IEntry = {
```

Comment:
```suggestion
        const resultAfterBroadcast: StorageTypes.IEntry = {
```

broadcast is one verb